### ISSUE TITLE
Allow body to json on PUT/POST/PATCH

### ISSUE THREAD

**Issue by ellen2212** [StefH/angular-odata-es5#94]
Hi,

thank you for your library, it is useful.
However we need post/put the json data to server and it didn't support.
![1589267943412](https://user-images.githubusercontent.com/18044804/81651360-35984200-9465-11ea-8049-4d5bd3422b43.jpg)

could you please export a function in ODataConfiguration.
Or other any way.
Thanks,
Ellen

**Comment by Sir-J** [comment]
Please, create an example, how are you want to use a new function.

**Comment by ellen2212** [comment]
Thank you for your reply.
I need a new function like this.
[example/odata.service.ts (function extractRequestDataFormat) ](https://github.com/ellen2212/angular-odata-es5/commit/76e8912b0101ec9efa08503648e1611d91b57858)
Thanks,
Ellen

**Comment by StefH** [comment]
Actually you want to be able to control how the data is serialized before it's send to the server?
At this moment, JSON.stringify is always used.

Correct?

**Comment by ellen2212** [comment]
Yes.Could you help us to fix it, or I create a new pull request.Thanks very much.

**Comment by ellen2212** [comment]
@StefH Do you have any advice to help us. Thanks very much.
